File: src/SchedulerMetrics.java
import java.util.*;

public class SchedulerMetrics {

    private double averageWaitingTime;
    private double averageTurnaroundTime;
    private double averageResponseTime;

    public SchedulerMetrics(List<Process> processes) {
        calculate(processes);
    }

    private void calculate(List<Process> processes) {
        double totalWaitingTime      = 0;
        double totalTurnaroundTime   = 0;
        double totalResponseTime     = 0;
        int n = processes.size();

        for (Process p : processes) {
            totalWaitingTime    += p.getWaitingTime();
            totalTurnaroundTime += p.getTurnaroundTime();
            totalResponseTime   += p.getResponseTime();
        }

        this.averageWaitingTime    = totalWaitingTime    / n;
        this.averageTurnaroundTime = totalTurnaroundTime / n;
        this.averageResponseTime   = totalResponseTime   / n;
    }

    public double getAverageWaitingTime() {
        return averageWaitingTime;
    }

    public double getAverageTurnaroundTime() {
        return averageTurnaroundTime;
    }

    public double getAverageResponseTime() {
        return averageResponseTime;
    }
}

File: web/SchedulerMetrics.js
class SchedulerMetrics {

    constructor(processes) {
        this.averageWaitingTime = 0;
        this.averageTurnaroundTime = 0;
        this.averageResponseTime = 0;
        
        this.calculate(processes);
    }

    calculate(processes) {
        let totalWaitingTime      = 0;
        let totalTurnaroundTime   = 0;
        let totalResponseTime     = 0;
        let n = processes.length;

        for (let p of processes) {
            totalWaitingTime    += p.getWaitingTime();
            totalTurnaroundTime += p.getTurnaroundTime();
            totalResponseTime   += p.getResponseTime();
        }

        this.averageWaitingTime    = totalWaitingTime    / n;
        this.averageTurnaroundTime = totalTurnaroundTime / n;
        this.averageResponseTime   = totalResponseTime   / n;
    }

    getAverageWaitingTime() {
        return this.averageWaitingTime;
    }

    getAverageTurnaroundTime() {
        return this.averageTurnaroundTime;
    }

    getAverageResponseTime() {
        return this.averageResponseTime;
    }
}